pkg/controller/pod_test.go

```diff
@@ -616,7 +629,57 @@ var _ = Describe("Dynamic Attachment controller", func() {
 					}).Should(ConsistOf(defaultNet))
 				})
 			})
+			When("an attachment is added with update status failed", func() {
+				JustBeforeEach(func() {
+					var err error
+					_, err = k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						updatePodSpec(pod, networkName, networkToAdd),
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("an `AddedInterface` and then a `FailedRemovingInterface` event are seen in the event recorded ", func() {
+
+					expectedError := errors.New("someerror")
+					k8sClient.PrependReactor("update", "pods", func(_ k8stesting.Action) (bool, runtime.Object, error) {
+						return true, nil, expectedError
+					})
+
+					expectedEventPayload := fmt.Sprintf(
+						"Normal AddedInterface pod [%s]: added interface %s to network: %s",
+						annotations.NamespacedName(namespace, podName),
+						"net1",
+						networkToAdd,
+					)
 
+					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
+
+					expectedEventPayload = fmt.Sprintf(
+						"Warning FailedRemovingInterface pod [%s]: failed removing interface %s from network: %s",
+						annotations.NamespacedName(namespace, podName),
+						"net1",
+						networkToAdd,
+					)
+
+					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
+				})
+
+				It("the pod network-status dosen't change", func() {
```

Comment:
I am not sure if this is required as the status update has been prevented, so this might test Kubernetes.